seqr/views/apis/summary_data_api.py

```diff
@@ -172,6 +183,140 @@ def bulk_update_family_analysed_by(request):
     })
 
 
+def _load_aip_data(data, user):
+    category_map = data['metadata']['categories']
+    results = data['results']
+
+    family_id_map = dict(Individual.objects.filter(
+        family__project__in=get_internal_projects(), individual_id__in=results.keys(),
+    ).values_list('individual_id', 'family_id'))
+    missing_individuals = set(results.keys()) - set(family_id_map.keys())
+    if missing_individuals:
+        raise ErrorsWarningsException([f'Unable to find the following individuals: {", ".join(sorted(missing_individuals))}'])
+
+    all_variant_ids = set()
+    family_variant_data = {}
+    for family_id, variant_pred in results.items():
+        family_variant_data.update({
+            (family_id_map[family_id], variant_id): pred for variant_id, pred in variant_pred.items()
+        })
+        all_variant_ids.update(variant_pred.keys())
+
+    saved_variant_map = {
+        (v.family_id, v.variant_id): v
+        for v in SavedVariant.objects.filter(family_id__in=family_id_map.values(), variant_id__in=all_variant_ids)
+    }
+
+    new_variants = set(family_variant_data.keys()) - set(saved_variant_map.keys())
+    if new_variants:
+        saved_variant_map.update(_search_new_saved_variants(new_variants, user))
+
+    aip_tag_type = VariantTagType.objects.get(name=AIP_TAG_TYPE, project=None)
+    existing_tags = {
+        tuple(t.saved_variant_ids): t for t in VariantTag.objects.filter(
+            variant_tag_type=aip_tag_type, saved_variants__in=saved_variant_map.values(),
+        ).annotate(saved_variant_ids=ArrayAgg('saved_variants__id', ordering='id'))
+    }
+
+    today = datetime.now().strftime('%Y-%m-%d')
+    update_tags = []
+    num_new = 0
+    for key, pred in family_variant_data.items():
+        metadata = {category: {'name': category_map[category], 'date': today} for category in pred['categories']}
+        updated_tag = _set_aip_tags(
+            key, metadata, pred['support_vars'], saved_variant_map, existing_tags, aip_tag_type, user,
+        )
+        if updated_tag:
+            update_tags.append(updated_tag)
+        else:
+            num_new += 1
+
+    VariantTag.bulk_update_models(user, update_tags, ['metadata'])
+
+    summary_message = f'Loaded {num_new} new and {len(update_tags)} updated AIP tags for {len(family_id_map)} families'
+    safe_post_to_slack(
+        SEQR_SLACK_DATA_ALERTS_NOTIFICATION_CHANNEL,
+        f'{summary_message}:\n```{", ".join(sorted(family_id_map.keys()))}```',
+    )
+
+    return create_json_response({
+        'info': [summary_message],
+    })
+
+
+def _search_new_saved_variants(family_variant_ids, user):
```

Comment:
I'm used to adding type hints for fn parameters and return values in Python for the sake of readability and maintainability. We also had a lot of teams making changes in other team's repo, so it was easier for someone new to understand what's happening in the codebase.

Are type hints something you've considered? 

I've seen type hints with custom types that are very verbose `_search_new_saved_variants(family_variant_ids: List[Tuple[ FamilyGuID, VariantGuID]], user: User): -> Dict[Tuple[ FamilyGuID, VariantGuID], Variant]]`

and ones that use the basic data types
`_search_new_saved_variants(family_variant_ids: list, user) -> dict:`

Historically I tend to write type hints that feel too verbose to me, taking up too much space in the function signature and requiring extra code and tests, so I'm open to hearing your thoughts.

Reply:
We have considered them, they are in the category of "it would be nice if we had them but adding them everywhere is hard". But probably a good idea to start adding them to new functions like this one!

Reply:
added!